shared/src/main/scala/effekt/core/Transformer.scala

```diff
@@ -235,11 +236,28 @@ class Transformer extends Phase[Module, core.ModuleDecl] {
     StateEffect(eff, get, put)
   }
 
+  def freshTmpFor(e: source.Tree)(implicit C: TransformerContext): Tmp = {
+    val x = Tmp(C.module)
+    C.typeOf(e) match {
+      case Some(t / _) => C.assignType(x, t)
+      case _           => C.abort("Internal Error: Missing type of source expression.")
+    }
```

Comment:
You could replace this with:

```
val (tpe / eff) = C.typeOf(e).getOrElse { C.abort("Internal Error: Missing type of source expression.") }
C.assignType(x, tpe)
```
but it is mostly personal taste. In your version the side effect `assignType` is a tiny bit hidden.